spec/services/form_validator_spec.rb

```diff
@@ -3,196 +3,125 @@
-          @params.merge!(test: ";) hey there")
-        end
+      it "with missing the TLD" do
+        params.merge!(email: "neal@sumofus")
+        expect(subject).to_not be_valid
       end
 
-      describe "email" do
-
-        before :each do
-          element.update_attributes({data_type: "email"})
-        end
-
-        it "is a valid email" do
-          @params.merge!(test: "neal@sumofus.org")
-        end
-
-        it "is a valid email with multiple dots" do
-          @params.merge!(test: "neal.donnelly@cycles.cs.princeton.edu")
-        end
-
-        it "is empty and not required" do
-          @params.merge!({})
-        end
+      it "with two @ symbols" do
+        params.merge!(email: "this@that@other.com")
+        expect(subject).to_not be_valid
       end
+    end
+  end
 
-      describe "phone" do
+  context "with phone as data_type" do
+    let(:element) { create :form_element, :phone, form: form }
+    let(:params){ {form_id: element.form_id, phone: '00 2323 12345' } }
 
-        before :each do
-          element.update_attributes({data_type: "phone"})
-        end
+    context "is valid" do
+      it "with regular numbers" do
+        expect(subject).to be_valid
+      end
 
-        it "is all numbers" do
-          @params.merge!(test: "123456790")
-        end
+      it "has spaces, dashes, pluses, and parentheses" do
+        params.merge!(phone: "+1 (413)-555-1234")
+        expect(subject).to be_valid
+      end
+    end
 
-        it "has spaces, dashes, pluses, and parentheses" do
-          @params.merge!(test: "+1 (413)-555-1234")
-        end
+    context "is invalid" do
+      it "if too short" do
+        params.merge!(phone: "12345")
+        expect(subject).to_not be_valid
+      end
 
-        it "is empty and not required" do
-          @params.merge!({})
-        end
+      it "with invalid special characters" do
+        params.merge!(phone: "(+) -- (+)")
+        expect(subject).to_not be_valid
       end
+    end
+  end
 
-      describe "country" do
+  context "with country as data_type" do
+    let(:element) { create :form_element, :country, form: form }
+    let(:params){ { form_id: element.form_id, country: 'FR' } }
 
-        before :each do
-          element.update_attributes({data_type: "country"})
-        end
+    it "is valid" do
+      expect(subject).to be_valid
+    end
 
-        it "is all a known country code" do
-          @params.merge!(test: "GB")
-        end
+    context "is invalid" do
+      it "with full country name" do
+        params.merge!(country: "France")
+        expect(subject).to_not be_valid
+      end
 
-        it "is empty and not required" do
-          @params.merge!({})
-        end
+      it "with a number" do
+        params.merge!(country: "33")
+        expect(subject).to_not be_valid
       end
 
+      it "as lowercase" do
+        params.merge!(country: "fr")
+        expect(subject).to_not be_valid
+      end
     end
-
   end
-
 end
+
```

Comment:
@NealJMD I've changed a little the spec file for `form_validator` (your specs are all there). I had some failing specs, and because the expectation statements were within `after` blocks, rspec was giving me wrong line numbers for the specs in question. I know it's a little extra code, but is it ok if we have an expectation within each spec?